whitelist.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"net"
+)
+
+type IPWhitelist struct {
+	ip4Whitelist *[]*ipNetMap
+	ip6Whitelist *[]*ipNetMap
+}
+
```

Comment:
It probably makes more sense to call this datastructure something like IPSet or IPMatcher or something. It could in theory be used also as a blacklist or some other IP-based selector.

Reply:
Also, if it's not going to expose any "public" members (e.g. `func (w* IPWhitelist) HasIP(ip net.IP)`), the type itself should be private, since it's not really usable outside of this package anyway.

Reply:
I agree, on both points